### DISCUSSION TITLE
MQTT - Motion Files

### DISCUSSION THREAD

**Discussion by rpkilgore** [roleoroleo/yi-hack-Allwinner-v2#403] [Q&A] [answered]
I'm attempting to write a script that will use the mqtt message que's to get the motion files, but I haven't been able to capture that message yet from the cam.  Anyone had any luck pulling the motion files from that topic?  I can see most all the other start and stop messages, but not the video file list.

I should mention I'm testing this on 2 r35gb cams.  Here is a screen shot where you can see the motion stop events, but not the motion_files.  Anyone else successfully get the motion files from mqtt?  The top screen is subscribed to all events, the bottom one is specifically subscribed to the motion_files.

![mqtt](https://user-images.githubusercontent.com/19828182/161447895-8adca5d8-726b-43df-b371-87cf0f69e42a.png)

**Comment by rpkilgore** [comment]
So after some tinkering I found that to get the mqtt messages to work then AI Human detection has to be turned off.  Then I started getting some messages. Yay! 

![mqtt](https://user-images.githubusercontent.com/19828182/161847272-753a1e5e-806e-469f-9f7d-6c62031d1355.png)

the motion_files topic didn't exactly give me what I was expecting, instead of the actual file names it appears to give a start and end time stamp.  And even more strange is it occasionally sends a 1969 time stamps.  From observation though this appears to happen on the motion start events, after the motion stop events it seems to reliably send a proper date/time stamp, but I guess that means I have to convert that to a valid filename?  I've also not9ced that it also gets the end time and start time swapped. 
Anyone else notice these quirks and has anyone else wrote any scripts to pull the vids from the camera using these events?

**Comment by roleoroleo** [comment]
You are right.
There is a problem in the section "files":
```
{
"start":"2022-04-07T15:00:09+0200",
"end":"2022-04-07T15:01:10+0200",
"files":[ ]
}
```

I have to check the code.

**Comment by roleoroleo** [comment, marked as answer]
I fixed the problem, next release...

**Reply by rpkilgore** [reply]
Awesome!!  Can't wait to test it out!  Thanks for taking a look at that and fixing it.  Does it now provide the full path to the file or just the record folder or just the mp4 file name itself?  I'm going to be contributing to your beer fund for this one! :)

**Reply by roleoroleo** [reply]
The path is the same: "directory/file.mp4".
I can't add the full path with the web address because the process isn't in the web context.
I don't know what is the name/ip of the cam.

**Reply by rpkilgore** [reply]
That's perfect! Thanks!  When will this be released?  Oh noticed you committed the changes, I'll update my repo, compile and do some testing..

**Reply by rpkilgore** [reply]
Just an update, so far in my testing with a y28ga this is working perfectly!

Thanks again @roleoroleo great work!

**Reply by roleoroleo** [reply]
Good news. I hope to release a new version tomorrow.

**Reply by rpkilgore** [reply]
@roleoroleo  I used the web interface to do an upgrade on 2 r35gb's to the latest release so I could utilize the files mqtt, but it doesn't appear to have fixed it.. I'm still getting empty payloads.  Should I try doing a fresh install of the hack?  Might there be something in doing an upgrade that would not apply your fix?

**Reply by roleoroleo** [reply]
The patch is ok:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/commit/9cfe97ad89196519237091779dde3663f6d9e066
Try a fresh install, please.

**Reply by rpkilgore** [reply]
Sorry for the long delay, it took awhile before I had physical access to the cameras.  After lots of installs, upgrades, and fresh installs I can confirm the following:
Update works on the r30gb, and the y28ga...
Update does NOT work on the r35gb, or the y211ga (at firmware verion 9)
for those 2 models I'm still getting empty payloads for the filenames..

**Reply by roleoroleo** [reply]
What's the behavior in r35gb and y211ga?

**Reply by rpkilgore** [reply]
Same as initially reported.  Start and End time are populated but the files array is blank.
r35gb:
![r35gb](https://user-images.githubusercontent.com/19828182/173104599-268f5696-cb45-49f2-ae05-2a8c0200869b.png)
y211ga:
![y211ga](https://user-images.githubusercontent.com/19828182/173104640-3fc51ac0-dc8e-494e-8004-9a0bbace07f8.png)

**Reply by roleoroleo** [reply]
There is another commit:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/commit/006d6e9372401ae635491eb8f21806272a98d097

I will send a beta to test.

**Reply by roleoroleo** [reply]
Here it is:
[mqttv4.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/8884439/mqttv4.gz)

**Reply by rpkilgore** [reply]
Cool.  I will test this out as soon as I can.  Thanks!
